File: uni/src/api/tryonRechargeOrder.js
import { request } from '@/utils/request.js'

// 创建试衣币充值订单
export const createTryonRechargeOrder = (data) => {
  return request({
    url: '/tryonRechargeOrder/createTryonRechargeOrder',
    method: 'post',
    data,
  })
}

// 更新充值订单支付方式（仅待支付）
export const updateTryonRechargeOrderPayMethod = (data) => {
  return request({
    url: '/tryonRechargeOrder/updateTryonRechargeOrderPayMethod',
    method: 'post',
    data,
  })
}

// 提交充值订单付款确认（仅扫码支付且待支付）
export const submitTryonRechargeOrderPayment = (data) => {
  return request({
    url: '/tryonRechargeOrder/submitTryonRechargeOrderPayment',
    method: 'post',
    data,
  })
}

// 取消充值订单（仅待支付）
export const cancelTryonRechargeOrder = (id) => {
  return request({
    url: '/tryonRechargeOrder/cancelTryonRechargeOrder',
    method: 'post',
    params: { ID: id },
  })
}

// 获取我的充值订单详情
export const selfTryonRechargeOrder = (id) => {
  return request({
    url: '/tryonRechargeOrder/selfTryonRechargeOrder',
    method: 'get',
    params: { ID: id },
  })
}

// 获取我的充值订单列表
export const getMyTryonRechargeOrderList = (params = {}) => {
  return request({
    url: '/tryonRechargeOrder/getMyTryonRechargeOrderList',
    method: 'get',
    params,
  })
}

File: server/router/client/tryon_recharge_order.go
package client

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/gin-gonic/gin"
)

// TryonRechargeOrderRouter 试衣币充值订单路由
type TryonRechargeOrderRouter struct{}

// InitTryonRechargeOrderRouter 初始化试衣币充值订单路由
func (s *TryonRechargeOrderRouter) InitTryonRechargeOrderRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	_ = PublicRouter
	tryonRechargeOrderRouter := Router.Group("tryonRechargeOrder")
	tryonRechargeOrderApi := v1.ApiGroupApp.ClientApiGroup.TryonRechargeOrderApi
	{
		tryonRechargeOrderRouter.POST("createTryonRechargeOrder", tryonRechargeOrderApi.CreateTryonRechargeOrder)
		tryonRechargeOrderRouter.POST("updateTryonRechargeOrderPayMethod", tryonRechargeOrderApi.UpdateTryonRechargeOrderPayMethod)
		tryonRechargeOrderRouter.POST("submitTryonRechargeOrderPayment", tryonRechargeOrderApi.SubmitTryonRechargeOrderPayment)
		tryonRechargeOrderRouter.POST("cancelTryonRechargeOrder", tryonRechargeOrderApi.CancelTryonRechargeOrder)
		tryonRechargeOrderRouter.POST("confirmTryonRechargeOrderPayment", tryonRechargeOrderApi.ConfirmTryonRechargeOrderPayment)
		tryonRechargeOrderRouter.GET("selfTryonRechargeOrder", tryonRechargeOrderApi.SelfTryonRechargeOrder)
		tryonRechargeOrderRouter.GET("findTryonRechargeOrder", tryonRechargeOrderApi.FindTryonRechargeOrder)
		tryonRechargeOrderRouter.GET("getMyTryonRechargeOrderList", tryonRechargeOrderApi.GetMyTryonRechargeOrderList)
		tryonRechargeOrderRouter.GET("getTryonRechargeOrderList", tryonRechargeOrderApi.GetTryonRechargeOrderList)
	}
}
